tests/python/test_slowmo_fsdp.py

```diff
@@ -0,0 +1,378 @@
+        self.assertEqual(
+            slowmo_optim_dummy.averager.period, slowmo_optim.averager.period
+        )
+        self.assertEqual(slowmo_optim_dummy.averager.step, slowmo_optim.averager.step)
+
+        # make sure slowmo parameters were updated
+        self.assertEqual(slowmo_optim_dummy.slowmo_freq, slowmo_optim.slowmo_freq)
+        self.assertEqual(slowmo_optim_dummy.slowmo_factor, slowmo_optim.slowmo_factor)
+        self.assertEqual(slowmo_optim_dummy.slowmo_lr, slowmo_optim.slowmo_lr)
+
+        for _ in range(10):
+            self._train_step(inpt, fsdp_net_slowmo, slowmo_optim_dummy)
+
+        self.assertEqual(slowmo_optim_dummy.averager.step, 20)
+
+    @skip_if_lt_x_gpu(2)
+    def test_slowmo_optimizer_errors(self):
+        net = torch.nn.Linear(1, 3, bias=False)
+        with self.assertRaisesRegex(
+            ValueError, "Base optimizer is a required" " parameter."
+        ):
+            _ = slow_momentum_optimizer.SlowMomentumOptimizer(
+                base_optim=None, slowmo_freq=4, slowmo_factor=0.5, slowmo_lr=0.1
+            )
+
+        with self.assertRaisesRegex(
+            ValueError, "Invalid ``slowmo_freq`` parameter, must be a positive value."
+        ):
+            _ = slow_momentum_optimizer.SlowMomentumOptimizer(
+                base_optim=torch.optim.SGD(net.parameters(), lr=1e-2),
+                slowmo_freq=-3,
+                slowmo_factor=0.5,
+                slowmo_lr=0.1,
+            )
+
+        with self.assertRaisesRegex(
+            ValueError, "Invalid ``slowmo_factor`` parameter, must be non-negative."
+        ):
+            _ = slow_momentum_optimizer.SlowMomentumOptimizer(
+                base_optim=torch.optim.SGD(net.parameters(), lr=1e-2),
+                slowmo_freq=4,
+                slowmo_factor=-0.5,
+                slowmo_lr=0.1,
+            )
+
+        with self.assertRaisesRegex(
+            ValueError, "Invalid ``slowmo_lr`` parameter, must be non-negative."
+        ):
+            _ = slow_momentum_optimizer.SlowMomentumOptimizer(
+                base_optim=torch.optim.SGD(net.parameters(), lr=1e-2),
+                slowmo_freq=4,
+                slowmo_factor=0.5,
+                slowmo_lr=-0.1,
+            )
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize("sharding_strategy", [ShardingStrategy.NO_SHARD])
+    def test_slowmo_optimizer_buffert_init(self, sharding_strategy):
+
+        # default simple net has size=(1, 5)
+        fsdp_net_slowmo = self._init_fsdp(sharding_strategy)
+
+        slowmo_optim = self._get_slomo_optimizer(
+            base_optim=torch.optim.SGD(fsdp_net_slowmo.parameters(), lr=1e-2),
+            slowmo_freq=2,
+        )
+
+        for _, v in slowmo_optim.state.items():
+            self.assertEqual(
+                v["slow_momentum"], torch.tensor([0.0] * 5, device=self.rank)
+            )
+            self.assertEqual(v["prev_param"], torch.flatten(fsdp_net_slowmo.weight))
```

Comment:
this `flatten()` is to go from `[1, 5]` -> `[5]`?

maybe change the above to `torch.zeros([1, 5], device=self.rank)` instead?